packages/api/amplify_api_dart/lib/src/graphql/factories/graphql_request_factory.dart

```diff
@@ -216,14 +253,14 @@ class GraphQLRequestFactory {
     // e.g. { 'name': { 'eq': 'foo }}
     if (queryPredicate is QueryPredicateOperation) {
       final association = schema.fields?[queryPredicate.field]?.association;
-      // TODO(ragingsquirrel3): Change key logic when supporting CPK.
       final associatedTargetName =
           association?.targetNames?.first ?? association?.targetName;
       var fieldName = queryPredicate.field;
       if (queryPredicate.field ==
-          '${_lowerCaseFirstCharacter(schema.name)}.$idFieldName') {
-        // check for the IDs where fieldName set to e.g. "blog.id" and convert to "id"
-        fieldName = idFieldName;
+          '${_lowerCaseFirstCharacter(schema.name)}.$_defaultIdFieldName') {
+        // Check for the IDs where fieldName set to e.g. "blog.id" and convert to "id".
+        // This is only needed in older codegen files (without CPK).
+        fieldName = _defaultIdFieldName;
```

Comment:
nit: Discussed offline. Could leave some comments as to why each branch exists or in what scenario.

Reply:
tried to clarify comments, though not sure how much I added as there were already some comments